functions/template/README.md

```diff
@@ -0,0 +1,38 @@
+# some-function-name
+
+### Overview
+
+<!--mdtogo:Short-->
```

Comment:
Is this applicable to only Go functions? If so, move it to functions/go?

What about TS functions? Are we rationalizing this with:

https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/ts/create-kpt-functions/templates/func.mustache

Also, are usage docs for existing functions being changed to conform to this template?

Reply:
> Is this applicable to only Go functions? If so, move it to functions/go?

Moved it.

> What about TS functions?

GoogleContainerTools/kpt-functions-sdk#254

> Also, are usage docs for existing functions being changed to conform to this template?

Filed https://github.com/GoogleContainerTools/kpt/issues/1636